recordings-downloader/nodejs/index.js

```diff
@@ -0,0 +1,156 @@
+        });
+    }
+
+    // Pass conversationIds and modifiedMediaType objects to downloadRequest function and initiate the function.
+    downloadRequest(conversationIds, modifiedMediaType);
+}
+
+// Function that will format objects to be acceptable to postRecordingBatchRequest API. Also accepts modifiedMediaType array that will be used later.
+function downloadRequest (conversationIds, modifiedMediaType) {
+    const download = {
+        batchDownloadRequestList: conversationIds
+    };
+
+    recordingApi.postRecordingBatchrequests(download)
+        .then((job) => {
+            // Once the requests are completed. Generated result such as job will be sent to getRecordingBatchRequest together with modifiedMediaType. Also, initate the function.
+            getRecordingBatchRequest(job, modifiedMediaType);
+        })
+        .catch((err) => {
+            console.log('There was a failure calling postRecordingBatchrequests');
+            console.error(err);
+        });
+}
+
+// Function that will generate jobId that will be used to request for request of batch recordings.
+function getRecordingBatchRequest (job, modifiedMediaType) {
+    recordingApi.getRecordingBatchrequest(job.id)
+        .then((batchDownloadData) => {
+            // Call the function again after 5 seconds up until the result count is equals to expected result
+            if (batchDownloadData.expectedResultCount === batchDownloadData.resultCount) {
+                filterBatchDownloadData(batchDownloadData, modifiedMediaType);
+            } else {
+                console.log('Processed: ' + batchDownloadData.resultCount + '/' + batchDownloadData.expectedResultCount);
+                setTimeout(() => getRecordingBatchRequest(job, modifiedMediaType), 5000);
+            }
+        })
+        .catch((err) => {
+            console.log('There was a failure calling getRecordingBatchrequest');
+            console.error(err);
+        });
+}
+
+// Filter batchdownload data using array mapping.
+function filterBatchDownloadData (batchDownloadData, modifiedMediaType) {
+    for (result of batchDownloadData.results) {
+        console.log(JSON.stringify(result) + 'result');
+
+        const mapping = modifiedMediaType.find((val) => result.conversationId === val.conversationId);
+
+        console.log('mapping ' + JSON.stringify(mapping));
+
+        getExtensionFromMediaType(result, mapping);
+    }
+}
+// Assign extension for every filtered batchdownload data.
+function getExtensionFromMediaType (result, mapping) {
+    let ext = '';
+    if (mapping.mediaType === 'voice') {
+        ext = '.wav';
+    } else if (mapping.mediaType === 'chat' || mapping.mediaType === 'email') {
+        ext = '.txt';
+    } else {
+        ext = '';
+    }
+
+    createDirectory(result, ext);
+}
+// Generate directory for recordings that will be downloaded
+function createDirectory (result, ext) {
+    fs.mkdir('./' + (result.conversationId), function (err) {
```

Comment:
This is a potential race condition. Use `fs.mkdirSync(...)` to synchronously create the directory to ensure it's there before continuing on.